internal/provider/data_source_projects_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-IDentifier: MPL-2.0
+
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTFEProjectsDataSource_basic(t *testing.T) {
+	tfeClient, err := getClientUsingEnv()
+	if err != nil {
+		t.Fatal(err)
+	}
+	org, orgCleanup := createBusinessOrganization(t, tfeClient)
+	t.Cleanup(orgCleanup)
+	orgName := org.Name
+
+	description1 := "Project 1"
+	description2 := "Project 2"
+	description3 := "Project 3"
+
+	prj1 := createProject(t, tfeClient, org.Name, tfe.ProjectCreateOptions{
+		Name: "project1", Description: &description1,
+	})
+	prj2 := createProject(t, tfeClient, org.Name, tfe.ProjectCreateOptions{
+		Name: "project2", Description: &description2,
+	})
+	prj3 := createProject(t, tfeClient, org.Name, tfe.ProjectCreateOptions{
+		Name: "project3", Description: &description3,
+	})
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEProjectsDataSourceConfig(orgName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.#", "4"),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_projects.all", "projects.0.id"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.name", "Default Project"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.description", ""),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.id", prj1.ID),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.name", prj1.Name),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.description", prj1.Description),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.id", prj2.ID),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.name", prj2.Name),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.description", prj2.Description),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.id", prj3.ID),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.name", prj3.Name),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.description", prj3.Description),
```

Comment:
Since the ordering of the 4 projects is unknown checking description like this may not work. I suggest to remove description from the test entirely or setup a an array with all 4 descriptions and perform a check similar to my suggestion above.

Reply:
While a second array would enable checking if the description exists, it wouldn't test if it's assigned to the correct project. 

I feel, not checking the description attribute would make the test incomplete.